src/plugins/embeddable/server/plugin.ts

```diff
@@ -56,14 +57,15 @@ export class EmbeddableServerPlugin implements Plugin<object, object> {
       telemetry: this.telemetry,
       extract: this.extract,
       inject: this.inject,
+      migrate: this.migrate,
     };
   }
 
   public stop() {}
 
-  private telemetry = (state: EmbeddableInput, telemetryData: Record<string, any> = {}) => {
+  private telemetry = (state: EmbeddableStateWithType, telemetryData: Record<string, any> = {}) => {
     const enhancements: Record<string, any> = state.enhancements || {};
-    const factory = this.getEmbeddableFactory(state.id);
+    const factory = this.getEmbeddableFactory(state.type);
```

Comment:
We changed from `state.id` to `state.type` in `server/plugin.ts`, but not in `public/plugin.tsx`. Is this intentional?

Reply:
ups, thanks, will extract the functions into common place